.vsts-pr.yml

```diff
@@ -40,4 +40,13 @@ stages:
             -prepareMachine
           name: Build
           displayName: Build and run tests
-          condition: succeeded()
\ No newline at end of file
+          condition: succeeded()
+        - task: PublishTestResults@2
+          displayName: 'Publish test results'
+          inputs:
+            testResultsFormat: xUnit
+            testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
+            mergeTestResults: true
+            searchFolder: $(System.DefaultWorkingDirectory)
+            testRunTitle: sign unit tests - $(Agent.JobName)
+          condition: failed()
```

Comment:
Curious why not use `always()` or `succeededOrFailed()`? Would you want test results on success case so you have a baseline to compare to?

Reply:
Maybe, but the problem this change solves is that a PR build sometimes fails on a flaky test, but we don't know which test or the failure reason.  With this change, you can see that information ([example](https://dev.azure.com/dnceng-public/public/_build/results?buildId=811149&view=ms.vss-test-web.build-test-results-tab)) and work on test flakiness.

What information do you want from the baseline success case?  The tests don't log diagnostic information (e.g.:  `ITestOutputHelper`), so the baseline information they offer now is that they passed and execution time.

[The internal pipeline already uploads test results in failure or success.](https://github.com/dotnet/sign/blob/main/.vsts-ci.yml#L129)

Reply:
Ah, I didn't know about the other internal pipeline. 

Might be nice to be able to confirm a new test actual ran (and wasn't ignored by the test runner) and it would be consistent with the other pipeline. But I don't feel strongly. Feel free to ignore.

Reply:
I changed it to `succeededOrFailed()`.  Thanks for the feedback.